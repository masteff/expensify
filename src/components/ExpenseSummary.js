import React from 'react'
import {connect} from 'react-redux'
import selectExpenses from '../selectors/expenses'
import selectTotalExpenses from '../selectors/expenses-total'
import numeral from 'numeral'

export const ExpenseSummary = ({total, number}) => {
    const expenseWord = number === 1 ? 'expense' : 'expenses'
    return (
        <div>
        <h2>Viewing {number} {expenseWord} with totalling of {numeral(total/100).format('$0,0.00')}</h2>
    </div>
    )
}
    

const mapStateToProps = (state) => ({
    total: selectTotalExpenses(selectExpenses(state.expenses,state.filters)),
    number: selectExpenses(state.expenses,state.filters).length
})

export default connect(mapStateToProps)(ExpenseSummary)