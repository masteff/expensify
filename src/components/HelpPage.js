import React from 'react';

const HelpPage = () => (
    <div>
        This is from my edit help component
    </div>
)

export default HelpPage