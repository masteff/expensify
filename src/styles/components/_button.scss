.button {
    background: $blue;
    border: none;
    padding: $s-size $m-size;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    text-decoration: none;
    line-height: 1;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888888;
}