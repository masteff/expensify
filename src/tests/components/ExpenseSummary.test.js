import React from 'react'
import {shallow} from 'enzyme'
import {ExpenseSummary} from '../../components/ExpenseSummary'

test('should redner Expense Summary with 1 expense coorectly', () => {
    const wrapper = shallow(<ExpenseSummary number={1} total={100}/>)
    expect(wrapper).toMatchSnapshot()
})

test('should redner Expense Summary with more expenses coorectly', () => {
    const wrapper = shallow(<ExpenseSummary number={5} total={100}/>)
    expect(wrapper).toMatchSnapshot()
})