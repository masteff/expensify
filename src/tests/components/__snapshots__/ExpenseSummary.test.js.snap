// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should redner Expense Summary with 1 expense coorectly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       with totalling of 
      <span>
        $1.00
      </span>
    </h1>
    <div
      className="page_header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should redner Expense Summary with more expenses coorectly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        5
      </span>
       
      expenses
       with totalling of 
      <span>
        $1.00
      </span>
    </h1>
    <div
      className="page_header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
